flask/app/routes.py

```diff
@@ -196,8 +220,8 @@ def mixin(entity: db.Model, json_mixin: Dict[str, Any], columns: List[str]) -> U
                 if not hasattr(type(column), str(value)):
                     allowed = [e.value for e in type(column)]
                     return f'"{field}" must be one of {allowed}'
-            setattr(entity, field, value)
-
+            if edit:
+                setattr(entity, field, value)
```

Comment:
I'm not sure if it can be applied to a model object but there is a way to do the validation without needing the result from a successful query (see 239e031)